2020/day3.clj

```diff
@@ -7,27 +7,39 @@
        (string/split-lines)))
 
 (defn next-location
-  [loc [right down]]
-  [(+ (first loc) right)
-   (+ (second loc) down)])
+  [[row col] [right down]]
+  [(+ row right)
+   (+ col down)])
 
 (defn locations-s
-  [height [right down]]
-  (take-while #(< (second %) height)
-              (map #(next-location [0 0] [(* right %) (* down %)])
-                   (range))))
+  [slope]
+  (let [right (first slope)
+        down  (second slope)]
+    (map (fn [n]
+           (next-location [0 0] [(* right n) (* down n)]))
```

Comment:
Still compressing - why?

Reply:
> Still compressing - why?

How was the new code?
```
(defn next-location-step
  [right down n]
  (next-location [0 0] [(* right n) (* down n)]))

(defn locations-s
  [slope]
  (let [right (first slope)
        down  (second slope)]
    (map (partial next-location-step right down)
         (range))))
```

Reply:
much better. The next-location should probably have a parameter per line